### PR TITLE
Enable UI sections based on feature enablement info from the server

### LINKED ISSUE #64: Support features endpoint to disable sections of the UI
See https://github.com/spring-cloud/spring-cloud-dataflow/issues/640

If the task support is disabled, the UI should not show task/job related components.

### PR DESCRIPTION
**Pull request by BoykoAlex**
Resolves #64

### CODE REVIEW

**Comment by BoykoAlex** [comment]
Took minimal amount of changes direction.
Let me know if it's best to follow `securityInfo` constant pattern for features enablement. That was my first solution, but blocking the UI until `/features` request is resolved seemed like not a good idea.

**Comment by ilayaperumalg** [comment]
Given the `dashboard` home page always redirects to the route `/streams/definitions`, we either need a different route (possibly `/apps`) or make the `features` info available before routes (which makes it look like having constant like `securityInfo`).

**Comment by ilayaperumalg** [comment]
We need to decide if we are ok with:
1) The `/apps` route can be the default in the UI page (since that feature is always available)
2) Fix the redirection based on the features condition as mentioned above.

**Comment by sabbyanandan** [comment]
Let's please default to `/apps` as the landing page.

**Comment by BoykoAlex** [comment]
I have made `/apps/apps` as the default URL

**Comment by BoykoAlex** [comment]
I have added the fix for direct navigation to `/streams` `/analytics`, `/tasks` and `/jobs`. Was rather trivial... A bit more complication on reviewing `routes.js`, but i think changes are obvious and easy to follow.

File: ui/app/index.html [hunk 1]
```diff
@@ -46,16 +46,16 @@
             <li ng-class="{ active: $state.includes('home.runtime') }">
                <a ui-sref="home.runtime.tabs.runtimeAppsList">Runtime</a>
             </li>
-            <li ng-class="{ active: $state.includes('home.streams') }">
+            <li ng-show="features.streamsEnabled" ng-class="{ active: $state.includes('home.streams') }">
```
Comment (ilayaperumalg) [inline ui/app/index.html]: The `navbar-brand` needs to be fixed:

> <a class="navbar-brand" ui-sref="home.jobs.tabs.definitions"><span></span></a>

will fix while merging.

**Comment by ilayaperumalg** [comment]
@BoykoAlex Thank you for the quick turnaround. The changes look great. merging.

**Comment by ilayaperumalg** [comment]
Squashed, rebased and merged as 3e7aa95237c888bde4c6ec8f8f32a230562b9ece

**Pull request closed without merging**